src/QMCWaveFunctions/Fermion/DiracMatrixComputeCUDA.hpp

```diff
@@ -217,19 +222,27 @@ class DiracMatrixComputeCUDA : public Resource
   }
 
 public:
-  DiracMatrixComputeCUDA() : Resource("DiracMatrixComputeCUDA")
+  DiracMatrixComputeCUDA()
+#if defined(ENABLE_OFFLOAD)
```

Comment:
I don't see how this class should even be visible to the compiler if ENABLE_OFFLOAD && ENABLE_CUDA aren't defined.  I did write the unit tests so you wouldn't need ENABLE_OFFLOAD defined, because why would you it's not offload code.  But is that even the reason for this?

Reply:
DiracMatrixInverter APIs uses offload allocators and allow me to switch to select the fallback. Then the API won't be compiled in your unit test/benchmark in a CUDA only build. Then I can only activate the base class when offload is on.

Reply:
Ok so the base class is only useful when offload is on.  It lets ENABLE_CUDA be on or off. But the other way around is sort of broken except for some unit testing, but that's not a new thing.